### ISSUE TITLE
[file configuration] Clarify what should happen for invalid environment variable substitution

### ISSUE THREAD

**Issue by pellared** [open-telemetry/opentelemetry-specification#3915]
**What are you trying to achieve?**

From: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution

> For example, `${API_KEY}` is valid, while `${1API_KEY}` and `${API_$KEY}` are invalid.

I think saying "invalid" does not make it clear how it should it be interpreted.

Should the parsing fail? _I do not think so,_

Should it not make the substitution and leave it as a string? _This is what I  propose._

Maybe we could rephrase it to e.g.

> For example, `${API_KEY}` is a environment variable substitution, while `${1API_KEY}` and `${API_$KEY}` are just strings.

CC @jack-berg

**Comment by pellared** [comment]
FY @marcalff

**Comment by marcalff** [comment]
> For example, `${API_KEY}` is a environment variable substitution, while `${1API_KEY}` and `${API_$KEY}` are just strings.

I agree.

**Comment by jack-berg** [comment]
Agree with the proposal. Let's open a PR and include an example as well.

**Comment by jack-berg** [comment]
I can sponsor, but not necessarily opening the PR myself. 

@pellared, @marcalff please review #4002, which proposes solving this by erroring instead of keeping a string. The related issue #3981 makes a good case for why erroring is preferred.
